### DISCUSSION TITLE
Create multiple sessions

### DISCUSSION THREAD

**Discussion by rubenvclceeic** [supabase/supabase#2869] [Questions] [answered]
How I can login with multiple users with Supebase c#?
At the moment, supebase is a sigleton, so you cannot use the supebase auth to auteticate multiple user. 

Do you know how to do thant on a  net core web aplication? 

Now when I use the supebase client the login is sharing on users, so the query fotward to database do with the last user login. Thats not correct.  

I don't know how to do multiple login.

**Comment by silentworks** [comment, marked as answer]
I'm not a C# developer but the library being a singleton shouldn't have anything to do with the session being shared. This would be down to how you have implemented the library more than anything, because each user would be starting a new session on the server inside your application.

**Reply by rubenvclceeic** [reply]
Web app on c# is  diferrent like xamarin o windows app. The app web run one time on server, so when you developer a singelton class don´t have to save curren session, because this class is use for all users. At the moment I'm developeing a supebase library on transient. This way allow multiple login users and it use the paradigm authorize of net core

**Reply by silentworks** [reply]
Every server-side app runs once on the server, however, session instances are per user using that app. This should not affect whether the class is a singleton or not. A singleton just means a single instance of that class is in use, it doesn't mean a single session is in use. It's probably best to open an issue on the c# library GitHub to see if they can better help with this.

**Reply by rubenvclceeic** [reply]
Ok, thank you @silentworks .
